pkg/spec/spec.go

```diff
@@ -300,9 +301,25 @@ func (config *CreateConfig) createConfigToOCISpec(runtime *libpod.Runtime, userM
 	blockAccessToKernelFilesystems(config, &g)
 
 	// RESOURCES - PIDS
-	if config.Resources.PidsLimit != 0 {
-		g.SetLinuxResourcesPidsLimit(config.Resources.PidsLimit)
-		addedResources = true
+	if config.Resources.PidsLimit > 0 {
+		// if running on rootless on a cgroupv1 machine, pids limit is
+		// not supported.  If the value is still the default
+		// then ignore the settings.  If the caller asked for a
+		// non-default, then try to use it.
+		setPidLimit := true
+		if rootless.IsRootless() {
+			cgroup2, err := cgroups.IsCgroup2UnifiedMode()
+			if err != nil {
+				return nil, err
+			}
+			if !cgroup2 && config.Resources.PidsLimit == sysinfo.GetDefaultPidsLimit() {
```

Comment:
Should we just handle this in `create.go` where we handle `cgroups=disabled`? Seems easier to just unset it there if we're rootless and CGroups V1.

Reply:
No because of podman-remote.  On a MAC or windows or remote client you are not going to know if the server is running on a rootless cgroupv1 system.  It is kind of a corner case, but the the CI/CD system would not pass without this change.

Reply:
Hmm. I don't like that we could be discarding user input here (`--pids-limit=4096` would get turned to 0 on rootless V1), but you're right that doing it in the frontend doesn't work for remote.